### PR TITLE
Major changes to filter_reads

### PR DESCRIPTION
**Pull request by polyatail**
This pull request addresses #79 by:

* Ignoring taxonomic assignments not passing filter
* Letting user override this functionality by passing --include-lowconf

In addition, this PR might break existing code by:

* Producing output that's different because of ignoring assignments not passing filter
* Renaming filter_reads to subset_reads

### CODE REVIEW

File: tests/test_scripts.py [hunk 1]
```diff
@@ -1,30 +1,43 @@
...
 
 
-@pytest.mark.parametrize('paired,split_pairs,with_children,exclude_reads', [
+@pytest.mark.parametrize('paired,split_pairs,with_children,exclude_reads,include_lowconf', [
     # single end
-    (False, False, False, False),
-    (False, False, True, False),  # --with-children
-    (False, False, True, False),  # --exclude-reads
-    (False, False, True, True),   # --with-children --exclude-reads
+    (False, False, False, False, False),
```
Comment (boydgreenfield) [inline tests/test_scripts.py]: This is insane, but OK 😱

File: onecodex/scripts/subset_reads.py [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+@click.option('-r', '--reverse', type=click.Path(),
+              help='The reverse (R2) read file, optionally.')
+@click.option('--validate/--no-validate', is_flag=True, default=True,
+              help='By default, validate FASTQ records (slow). Disable to drastically increase speed')
+@click.option('--with-children', default=False, is_flag=True,
+              help='Match child taxa of those given with -t (e.g., all strains of E. coli)')
+@click.option('--split-pairs', default=False, is_flag=True,
+              help='By default, if either read in a pair matches, both will match. Choose this '
+                   'option to consider each paired-end read separately. Resulting files may *not* '
+                   'have the same number of reads!')
```
Comment (boydgreenfield) [inline onecodex/scripts/subset_reads.py]: Let's call it `--subset-pairs-independently` and have the following help message:

```
By default, if either read in a pair matches, both will be retained in the subset file. With this option, R1 and R2 files will be evaluated independently. Note that the subset output FASTQs are *not* guaranteed to have the same number of reads!
```

Comment (polyatail) [reply onecodex/scripts/subset_reads.py]: Changed as requested.

File: tests/test_scripts.py (hunk 1 above)
Comment (polyatail) [reply tests/test_scripts.py]: It's just well-tested.

**Pull request merged**